content/ru/guide/v11/typescript.md

```diff
@@ -0,0 +1,566 @@
+    // и обеспечивает автоматическое заполнение строк типов.
+    case 'reset':
+      return initialState;
+    case 'increment':
+      return { count: state.count + 1 };
+    case 'decrement':
+      return { count: state.count - 1 };
+    default:
+      return state;
+  }
+}
+```
+
+Когда мы используем функцию редуктора в `useReducer`, мы выводим несколько типов и выполняем проверки типов для переданных аргументов.
+
+```tsx
+function Counter({ initialCount = 0 }) {
+  // TypeScript гарантирует, что редуктор имеет максимум два аргумента и что начальное состояние имеет тип Statetype.
+  // Более того:
+  // — state имеет тип StateType
+  // — dispatch это функция для отправки ActionType
+  const [state, dispatch] = useReducer(reducer, { count: initialCount });
+
+  return (
+    <>
+      Счётчик: {state.count}
+      {/* TypeScript гарантирует, что отправленные действия имеют тип ActionType. */}
+      <button onClick={() => dispatch({ type: 'reset' })}>Сброс</button>
+      <button onClick={() => dispatch({ type: 'increment' })}>+</button>
+      <button onClick={() => dispatch({ type: 'decrement' })}>-</button>
+    </>
+  );
+}
+```
+
+Единственная необходимая аннотация находится в самой функции редуктора. Типы `useReducer` также гарантируют, что возвращаемое значение функции редуктора имеет тип `StateType`.
+
+## Расширение встроенных типов JSX
+
+У вас могут быть [пользовательские элементы](/guide/v11/web-components), которые вы хотели бы использовать в JSX, или вы можете добавить дополнительные атрибуты ко всем элементам HTML для работы с определённой библиотекой. Для этого вам нужно расширить интерфейсы `IntrinsicElements` или `HTMLAttributes` соответственно, чтобы TypeScript знал об этом и мог предоставить корректную информацию о типе.
+
+### Расширение `IntrinsicElements`
+
+```tsx
+function MyComponent() {
+  return <loading-bar showing={true}></loading-bar>;
+  //      ~~~~~~~~~~~
+  //   💥 Ошибка! Свойство 'loading-bar' не существует для типа 'JSX.IntrinsicElements'.
+}
+```
+
+```tsx
+// global.d.ts
+
+declare global {
+  namespace preact.JSX {
+    interface IntrinsicElements {
+      'loading-bar': { showing: boolean };
+    }
+  }
+}
+
+// Этот пустой экспорт очень важен! Это говорит TS, что нужно рассматривать это как модуль
+export {}
+```
+
+### Расширение `HTMLAttributes`
+
+```tsx
+function MyComponent() {
+  return <div custom="foo"></div>;
+  //          ~~~~~~
+  //       💥 Ошибка! Тип '{ custom: string; }' не может быть присвоен типу 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'.
+  //                   Свойство 'custom' не существует для типа 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'.
+}
+```
+
+```tsx
+// global.d.ts
+
+declare global {
+  namespace preact.JSX {
+    interface HTMLAttributes {
+      custom?: string | undefined;
+    }
+  }
+}
+
+// Этот пустой экспорт очень важен! Это говорит TS, что нужно рассматривать это как модуль
+export {}
+```
```

Comment:
I think you might've merged two separate subsections here, see [the English docs](https://github.com/preactjs/preact-www/blob/15a1971f460fa27a04f91a1127db1cd4996f0f92/content/en/guide/v11/typescript.md?plain=1#L552-L610)? The use of `preact.JSX` is also outdated I believe.